storage/innobase/os/os0file.cc

```diff
@@ -3031,12 +3038,26 @@ os_aio_linux_handler(
 void
 AIO::os_aio_dispatch_read_array_submit_low(
 	bool acquire_mutex MY_ATTRIBUTE((unused)))
+{
+	os_aio_dispatch_read_array_submit_low_for_array(acquire_mutex, s_reads);
+	if (s_ibuf != NULL) {
```

Comment:
is the s_ibuf != null check required? If yes, isn't that true for other array type s_reads?

Reply:
Yes, this is clearly visible in AIO::start() line 7159.

For me it makes sense, as it is read only we do not need ibuf optimization.

Reply:
this part, I saw that. we dont initliaze the ibuf segment. My question is how buf_read_linear() and random() know that it shouldn't buffer reads on ibuf segment?

Reply:
srv_read_only_mode value is pushed down from fil0fil.cc _fil_io() function down to AIO::select_slot_array() where we have 
![image](https://user-images.githubusercontent.com/50845928/86466037-2e8a0380-bd33-11ea-878e-0d9c1c0b0fe4.png)

Reply:
Thanks!